modules/generic/src/main/scala/pureconfig/generic/error/UnexpectedValueForFieldCoproductHint.scala

```diff
@@ -0,0 +1,17 @@
+package pureconfig.generic.error
+
+import com.typesafe.config.{ ConfigRenderOptions, ConfigValue }
+import pureconfig.error.FailureReason
+
+/**
+ * A failure reason given when an unknown value was found in the discriminating field of a config value, when using a
+ * `FieldCoproductHint`.
+ *
+ * @param value the value found in the discriminating field of a config value
+ */
+final case class UnexpectedValueForFieldCoproductHint(value: ConfigValue) extends FailureReason {
+  def description =
+    s"Unexpected value ${value.render(ConfigRenderOptions.concise())} found. You might have a misconfigured " +
```

Comment:
I love this error message. Great job.

Do you think it'd be worth adding the type of the ADT/sealed family?

Reply:
Sorry! I missed this message. I think we can add the type of the sealed family here. We need to be careful, though, because this error will be scoped under the `type` field (or whatever the `key` of `FieldCoproductHint` is), so we need to be explicit that we're not trying to read that type from that field.

If we opt to add the type of the sealed family to this `FailureReason`, I think we should also do it for `NoValidCoproductChoiceFound`. I think doing so will complicate a bit the current implementation for deriving enumerations because we currently don't know the original type when we're deriving an `EnumerationConfigReaderBuilder` for the generic representation. Taking that into account, I'd rather leave that for another PR to avoid this getting too big.

Reply:
No worries.